### ISSUE TITLE
Update first-plugin example to best practices

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/docs#440]
## 🧰 Task

Currently we still recommend a setup.cfg and pyproject.toml
Probably best to switch to using just pyproject.toml and maybe a /src dir
https://learn.scientific-python.org/development/

**Comment by DragaDoncila** [comment]
I agree. I don't want to add a BUNCH of detail about python packaging etc., because we really want folks to focus on the plugin aspects, not the general python aspects. However, now that there is an actual blessed pathway with just `pyproject.toml`, a `src/` dir, and `build`, I think that should be what we demo.

**Comment by psobolewskiPhD** [comment]
Bumping this, we should ensure that there is parity between the documentation and the template repo:
https://github.com/napari/napari-plugin-template

And we should make sure we link to up-to-date guides on python packaging for people who want to dig more.

**Comment by TimMonko** [comment]
It would be wise to mostly direct this tutorial to point to python structuring found in napari-plugin-template. That way we don't have instructions go out of sync. We can move some of the 'reasoning' to the plugin template as well.

**Comment by psobolewskiPhD** [comment]
I'm of two minds there. I think this guide is supposed to keep it simple and be a from scratch example. So just pointing to the template repo isn't great? But for sure we should have one recommended packaging setup for plugins.

**Comment by DragaDoncila** [comment]
I think it's important to keep this guide to the bare bones "from scratch" example as a teaching tool. I have frequently referred to it on that basis, and used it for teaching purposes. I think the guide should refer to the `napari-plugin-template` as a "batteries included" template. 

I think updating to `pyproject.toml` + `python -m build` + `src` dir is the way to go.

**Comment by psobolewskiPhD** [comment]
We discussed this in a lead-up-to-SciPy-2025 meeting today. Updating this before the workshop is a high priority and we agreed that we want to keep it a simple plugin from scratch.
